lib/net.c

```diff
@@ -885,6 +890,51 @@ check_port_offloads(struct gatekeeper_if *iface,
 	if (iface->mtu > RTE_MBUF_DEFAULT_BUF_SIZE)
 		port_conf->txmode.offloads |= DEV_TX_OFFLOAD_MULTI_SEGS;
 
+	for (i = 0; i < iface->num_ports; i++) {
+		struct rte_eth_rss_conf cur_rss_conf;
+		uint16_t port_id = iface->ports[i];
+		int ret = rte_eth_dev_rss_hash_update(port_id, &rss_conf);
+		if (ret == 0)
+			continue;
+
+		ret = rte_eth_dev_rss_hash_conf_get(port_id, &cur_rss_conf);
+		if (ret < 0)
+			return ret;
```

Comment:
Since returning an error here signals that Gatekeeper does not support a critical hardware capability, we should have an error message describing what capability is missing (here and in the error conditions below).